### PR TITLE
Feature/add schema flatten

### PR DESCRIPTION
**Pull request by yruslan**
* Add support shotr names for schema flattening
* Cover the method with tests

@Zejnilovic, Thanks for the contribution to Cobrix

### CODE REVIEW

File: spark-cobol/src/main/scala/za/co/absa/cobrix/spark/cobol/utils/SparkUtils.scala [hunk 1]
```diff
@@ -70,49 +73,102 @@ object SparkUtils {
...
 
-    def flattenStructArray(path: String, structField: StructField, arrayType: ArrayType): Unit = {
+    /**
+      * For an array of arrays of arrays, ... get the final element type at the bottom of the array
+      *
+      * @param arrayType An array data type from a Spark dataframe schema
+      * @return A non-array data type at the bottom of array nesting
+      */
+    @tailrec
+    def getDeepestArrayType(arrayType: ArrayType): DataType = {
```
Comment (Zejnilovic) [inline spark-cobol/src/main/scala/za/co/absa/cobrix/spark/cobol/utils/SparkUtils.scala]: Did you decide to keep this in even though you aren't using it?

File: spark-cobol/src/main/scala/za/co/absa/cobrix/spark/cobol/utils/SparkUtils.scala [hunk 2]
```diff
@@ -70,49 +73,102 @@ object SparkUtils {
...
+            val newFieldName = getNewFieldName(s"$newFieldNamePrefix")
+            fields += expr(s"$path[$i]").as(newFieldName)
+            stringFields += s"""expr("$path`[$i] AS `$newFieldName`")"""
         }
         i += 1
       }
     }
 
+    def flattenNormalArray(path: String, fieldNamePrefix: String, structField: StructField, arrayType: ArrayType): Unit = {
+      flattenStructArray(path, fieldNamePrefix, structField, arrayType)
```
Comment (Zejnilovic) [inline spark-cobol/src/main/scala/za/co/absa/cobrix/spark/cobol/utils/SparkUtils.scala]: This is a one line method. If you don't have future plans with it, we can omit.

File: spark-cobol/src/main/scala/za/co/absa/cobrix/spark/cobol/utils/SparkUtils.scala (hunk 1 above)
Comment (yruslan) [reply spark-cobol/src/main/scala/za/co/absa/cobrix/spark/cobol/utils/SparkUtils.scala]: Fair point, will remove that

**Comment by yruslan** [comment]
Good spot!

**Comment by yruslan** [comment]
Awesome, Thanks for the review!

**Pull request merged**